### PR TITLE
CUMULUS-3641 - Update Collections LIST endpoint to query Postgres basic 

### PR DESCRIPTION
**Pull request by Nnaga1**
**Summary:** Summary of changes

Addresses [CUMULUS-XX: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-XXX)

## Changes

* Detailed list or prose of changes
* ...

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: packages/api/endpoints/collections.js
Comment (charleshuang80) [inline packages/api/endpoints/collections.js:52]: So Jenny's comments mentioned keeping `includeStats`, and also if it is true it should still use ES for that. I'm not exactly sure what that should look like, but that doesn't seem to be happening here.

**Review by charleshuang80 — COMMENTED** [review]
Some comments/suggested changes. Mostly minor, but one might not be. Still have to go through the tests, so need to continue with that.

File: packages/db/src/search/field-mapping.ts
Comment (charleshuang80) [inline packages/db/src/search/field-mapping.ts:126]: are these from the api docs or from transferring from ES (or both)? just curious about where to verify this is coming from

File: packages/api/tests/endpoints/collections/list-collections.js
Comment (charleshuang80) [inline packages/api/tests/endpoints/collections/list-collections.js:128]: So based on what happens with includeStats, I'm thinking this may need to change a bit. If it needs ES, I might suggest creating a separate temporary test file, since all the setup is a bit different.

File: packages/api/tests/endpoints/collections/list-collections.js
Comment (charleshuang80) [inline packages/api/tests/endpoints/collections/list-collections.js:125]: In general, this test seems fine. However, I'm wondering if we need to do any of the permutations that are in the granules endpoint tests, like pagination and limit, and if the meta needs to be included. Maybe that will come with a later ticket, but it feels like we should have some of that for collections when we're done with it.

**Review by charleshuang80 — COMMENTED** [review]
review for first test, one more to go

File: packages/db/tests/search/test-CollectionSearch.js
Comment (charleshuang80) [inline packages/db/tests/search/test-CollectionSearch.js:112]: Since you have a `term search for date field` case below, and since granules has it broken out, maybe we can be more specific about the search terms in the description, and have each of them separate?

File: packages/db/tests/search/test-CollectionSearch.js
Comment (charleshuang80) [inline packages/db/tests/search/test-CollectionSearch.js:91]: I don't think it matters if these are combined or not, but for granules they are separated, so we should have both be consistent

File: packages/db/src/search/CollectionSearch.ts
Comment (charleshuang80) [reply packages/db/src/search/CollectionSearch.ts:3]: Ah, ok. That makes it a little confusing but I see why you can't change it.

File: packages/api/tests/endpoints/collections/list-collections.js
Comment (charleshuang80) [reply packages/api/tests/endpoints/collections/list-collections.js:152]: well, that is the understanding I have from the other conversation with Jenny.

File: packages/api/tests/endpoints/collections/list-collections.js
Comment (charleshuang80) [reply packages/api/tests/endpoints/collections/list-collections.js:148]: Hmm. That seems fair, but granules has it in both places. :thinking_face:

File: packages/db/src/search/CollectionSearch.ts
Comment (charleshuang80) [reply packages/db/src/search/CollectionSearch.ts:54]: So what I mean is, this is what is in GranuleSearch
```
if (infix) {
      [countQuery, searchQuery].forEach((query) => query.whereLike(`${this.tableName}.granule_id`, `%${infix}%`));
    }
    if (prefix) {
      [countQuery, searchQuery].forEach((query) => query.whereLike(`${this.tableName}.granule_id`, `${prefix}%`));
    }
```
I _think_ both are doing the same thing, but are implemented slightly differently, and we should just choose one.

File: packages/db/src/search/CollectionSearch.ts
Comment (charleshuang80) [reply packages/db/src/search/CollectionSearch.ts:3]: After sitting on this a bit, I think what would be helpful here is a comment somewhere saying that there is no `ApiCollectionRecord` because `CollectionRecord` is the type from the API, and has all the same fields (also 'that are needed', if it's the case that there are other fields but not needed for querying).

File: packages/api/tests/endpoints/collections/list-collections.js
Comment (charleshuang80) [reply packages/api/tests/endpoints/collections/list-collections.js:148]: Ok. Since it is covered in the other tests, I'm ok to leave it for now, and we can revisit hopefully with Jenny when we are wanting to merge the feature branch.

File: packages/api/endpoints/collections.js
Comment (Jkovarik) [inline packages/api/endpoints/collections.js:47]: Nit: we almost never use trace over .info.        A decision to start doing that should be deliberate

File: packages/api/endpoints/collections.js
Comment (Jkovarik) [inline packages/api/endpoints/collections.js:50]: Q: Why is this change needed?

Comment (Jkovarik) [inline packages/api/endpoints/collections.js:50]: Not to belabor the work, but we should enable @ts-check and get the typings right for this method, especially if adding if/else logic, per our ADRs.

Comment (Jkovarik) [reply packages/api/endpoints/collections.js:50]: I mean, we can toss out the if/else portion of that and the method should still be annotated/typed with @ts-check.

File: packages/api/endpoints/collections.js
Comment (Jkovarik) [reply packages/api/endpoints/collections.js:47]: Ok, sure, that comment applies to both PRs then - this is in theory a change to how we're logging.   We should broadcast (arch meting)  and decide to do it, else efforts outside this insular work won't be consistent with the approach you're choosing. 

It's minor and shouldn't block the PR (nit), but it's notable, particularly if we ever get around to log-level squelching. 

@jennyhliu

Comment (charleshuang80) [reply packages/api/endpoints/collections.js:47]: I will add a comment to the feature branch, so we can revisit when Jenny is back and when we are thinking of merging the feature branch.

File: packages/api/endpoints/collections.js
Comment (charleshuang80) [reply packages/api/endpoints/collections.js:50]: So you're saying we should add a `// @ts-check` somewhere near the top of this file?

Comment (Jkovarik) [reply packages/api/endpoints/collections.js:50]: @charleshuang80  yes, exactly, and then resolve the issues related to the *changing* code.   Fixing all of it is a thing we are doing as we're updating.

**Review by Jkovarik — CHANGES_REQUESTED** [review]
Please enable ts-check on the endpoints logic!

**Comment by Jkovarik** [comment]
> > Please enable ts-check on the endpoints logic!
> 
> I don't agree with this, this creates errors in virtually every other function when it does getEsClient() (put, del, etc.) except for list pretty much (the fix for that is easy), I don't think I should add something that makes me change work that isn't mine or even related to mine and doesn't really have much value outside of a temporary fix/check

Respectfully, it's required per our team ADRs, and something we should be doing in all of our other PRs.   It's not expected that we fix all of the errors noted, but all code in the scope of the updates should be properly type annotated.     
 
 We should have a team conversation about this - if the team wants to change that stance, we should make that a deliberate decision.  
 
 Until this is resolved this cannot be merged.

**Comment by Jkovarik** [comment]
@Nnaga1 the changes in cea3c14 and db45ecd look good re: the collection endpoint typings.   Thank you!

File: packages/db/src/search/CollectionSearch.ts
Comment (charleshuang80) [reply packages/db/src/search/CollectionSearch.ts:59]: Ok, to be more clear, what I'm saying is, either change GranuleSearch so it looks like
```
if (infix) {
  countQuery.whereLike(`${this.tableName}.granule_id`, `%${infix}%`);
  searchQuery.whereLike(`${this.tableName}.granule_id`, `%${infix}%`);
}
```
and do the same for prefix

OR
change this file so it looks like
```
if (infix) { 
  [countQuery, searchQuery].forEach((query) => query.whereLike(`${this.tableName}.name`, `%${infix}%`)); 
}
```
and do the same for prefix

File: packages/db/tests/search/test-CollectionSearch.js
Comment (charleshuang80) [inline packages/db/tests/search/test-CollectionSearch.js:116]: this is really minor, and if you want to push off into the feature branch then fine, but I think you should try and order these tests to align with the ordering in test-GranuleSearch. That way comparisons and additions/subtractions of the tests will be easier to deal with.

Comment (charleshuang80) [reply packages/db/tests/search/test-CollectionSearch.js:116]: you can also move around the tests in GranuleSearch to get the ordering to match better if that helps

File: packages/db/tests/search/test-CollectionSearch.js
Comment (charleshuang80) [inline packages/db/tests/search/test-CollectionSearch.js:250]: should this description say collection field not granule field?

File: packages/db/src/search/field-mapping.ts
Comment (charleshuang80) [inline packages/db/src/search/field-mapping.ts:126]: Sorry, keep forgetting to add this comment. The feature branch has a test file `packages/db/tests/search/test-field-mapping.js`. As you are adding some field mappings, do you not need to add any tests to that file for these?

File: packages/db/tests/search/test-CollectionSearch.js
Comment (charleshuang80) [inline packages/db/tests/search/test-CollectionSearch.js:214]: should this be collection instead of granule? or am I misunderstanding what the results are here?

File: packages/db/src/search/field-mapping.ts
Comment (charleshuang80) [reply packages/db/src/search/field-mapping.ts:126]: Ok, I missed the actual test before, but it is in that file I referenced on line 104. I think that test should include some of these fields you added, like report_to_ems, sample_file_name, and url_path

**Review by charleshuang80 — APPROVED** [review]
looks good! sorry to be a stickler for small things, but thanks for making the changes. I think they'll help with maintainability and when others have to deal with the code.

File: example/spec/parallel/createReconciliationReport/CreateReconciliationReportSpec.js
Comment (jennyhliu) [inline example/spec/parallel/createReconciliationReport/CreateReconciliationReportSpec.js:276]: We should add support for `_id__in` for collection db query.

**Pull request merged**
